amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/reports/designer/validators/ReportHierarchyValidator.java

```diff
@@ -0,0 +1,64 @@
+package org.digijava.kernel.ampapi.endpoints.reports.designer.validators;
+
+import org.digijava.kernel.ampapi.endpoints.errors.ApiErrorMessage;
+import org.digijava.kernel.ampapi.endpoints.reports.designer.ReportColumn;
+import org.digijava.kernel.ampapi.endpoints.reports.designer.ReportColumnProvider;
+import org.digijava.kernel.ampapi.endpoints.reports.designer.ReportProfile;
+import org.digijava.kernel.ampapi.endpoints.reports.designer.ReportRequest;
+import org.digijava.kernel.ampapi.endpoints.reports.designer.ReportType;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.digijava.kernel.ampapi.endpoints.reports.designer.ReportDesignerErrors.REPORT_INVALID_HIERARCHIES;
+import static org.digijava.kernel.ampapi.endpoints.reports.designer.ReportProfile.REPORT;
+import static org.digijava.kernel.ampapi.endpoints.reports.designer.ReportProfile.TAB;
+
+/**
+ * Validates that hierarchy ids are valid in report
+ *
+ * @author Viorel Chihai
+ */
+public class ReportHierarchyValidator implements ReportValidator {
```

Comment:
this condition is in the current report wizard
You cannot Select Primary and Secondary Sectors as hierarchies

Reply:
No, since the condition is no more available. YAHOO.amp.reportwizard.fundingGroups["incompatible_hierarchies"] is empty.
It is a legacy code that should be removed, but not as part of this ticket.

Reply:
https://github.com/devgateway/amp/commit/e30048b398901cb1226fecb7a883259e581a5732?branch=e30048b398901cb1226fecb7a883259e581a5732&diff=unified#diff-8f14c3de96f21f7e9d7d46a6976cf1e7f3f9a5d02e69a4787a43fc4beb0b71bf